### PR TITLE
Return constraint name

### PR DESCRIPTION
**Pull request by mbrunthaler**
## Todos
- [X] MIT compatible
- [x] Tests
- [ ] Documentation
- [X] Updated CHANGES.rst

### CODE REVIEW

**Comment by graingert** [comment]
could do with some tests :)

File: tests/rs_sqla_test_utils/models.py [hunk 1]
```diff
@@ -102,6 +112,18 @@ class ReflectionForeignKeyConstraint(Base):
     )
 
 
+class ReflectionNamedForeignKeyConstraint(Base):
+    __tablename__ = 'reflection_named_fk_constraint'
+    col1 = sa.Column(sa.Integer(),
+                     sa.ForeignKey('reflection_unique_constraint.col1',
+                                   name="reflection_named_fk_constraint__fk"),
+                     primary_key=True)
```
Comment (graingert) [inline tests/rs_sqla_test_utils/models.py]:
```python
col1 = sa.Column(
    sa.Integer(),
    sa.ForeignKey(
        'reflection_unique_constraint.col1', 
        name="reflection_named_fk_constraint__fk"
    ),
    primary_key=true,
)
```

**Comment by graingert** [comment]
@mbrunthaler neat, what's this actually for?

File: tests/conftest.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def database_name_generator():
             count=i,
         )
 
+
```
Comment (graingert) [inline tests/conftest.py]: y

**Comment by mbrunthaler** [comment]
I do need constraint names to apply migrations (drop / update constraint e.g. when the primary key columns change). sqlalchemy doesn't compile DROP PRIMARY KEY statements when no name is given.

**Comment by graingert** [comment]
@mbrunthaler cool, can you squash the commits plz

File: tests/conftest.py (hunk 2 above)
Comment (mbrunthaler) [inline tests/conftest.py]: `tests/conftest.py:34:1: E305 expected 2 blank lines after class or function definition, found 1` but I'm reverting this.

**Pull request merged**